Ability to generate docs for properties, enums and classes

Zaid-Ajaj opened the pull request.

ful-stackz commented on src/SharpCode/Extensions.cs:
```diff
@@ -46,6 +48,27 @@ public static string ToSourceCode(this Parameter parameter)
+            var blockParts = new[]
+            {
+                 "/// <summary>",
+                 "/// " + docs.Replace("\r\n", "\r\n/// ").Replace("\n", "\r\n/// ") ,
```
Can you please use `\n` (unix style) as a new line character to be consistent with the rest of the methods? Same goes for line `@68` - `return string.Join(...`

You can also shorten this to a single `.Replace()` by using a RegEx like `/\r?\n/g` to match all new lines in replace them all at once.

Zaid-Ajaj replied:
Fixed

ful-stackz commented on src/SharpCode/Structures.cs:
```diff
@@ -32,11 +32,12 @@ internal class Property
     }
 
-    internal class Parameter
+    internal class Parameter 
```
Can you please remove trailing whitespace 😁 FxCop Analyzers coming soon to take care of that, but not there yet :(

ful-stackz commented:
Hey, @Zaid-Ajaj! Thank you very much for the contribution 😍 Too bad you won't be getting a t-shirt for it 😆  I like the proposed changes very much! I do have some points for discussion though.

## `WithDocumentation`

Documentation sounds a bit too generic. What I mean is XML documentation accepts a lot more than a `<summary>`, ie. `<param>`, `<returns>`, `<remarks>` etc. As a consumer I feel like the name of the method doesn't give me _everything that I might expect_ from it. I see two ways to resolve that.

### Option 1: `DocumentationBuilder`

One way would be for `WithDocumentation()` to accept a `DocumentationBuilder`, which will expose XML docs related things:

```
DocumentationBuilder
  + WithSummary(summary: string): DocumentationBuilder
  + WithParam(name: string, description: string): DocumentationBuilder
  + WithReturns(description: string): DocumentationBuilder
  // etc
```

This way will allow us to expand on the documentation module later on. On the other side I really can't imagine how easy or hard it would be to generate the summary, let alone other XML tags. So my second options is...

### Option 2: Narrow down

The other way forward is to simply rename `WithDocumentation(docs: string)` to `WithSummary(summary: string)`. This keeps the exposed API small, provides better metadata only by reading the method/param name and you already built the functionality behind it.

I'd say go for this option, because I imagine `.WithSummary()` will be the most commonly generated documentation, so it is good to have a straightforward API for it. Later on if there is real need for extensive documentation we can add the `WithDocumentation(builder: DocumentationBuilder)` that I described above, so it's going to be an addition, without breaking existing stuff.

Zaid-Ajaj commented:
Hey Ivan, glad you like the small feature :smile: I agree that `WithDocumentation` sounds a bit too encompassing. I went with option 2 to narrow down and renamed it to `WithSummary`. This is because it is the easiest to implement and because it applies nicely to properties, enums and classes which don't have `<param>` or `<return>` blocks. 

I also changed the new line replacements to use `\n` like you suggested and remove the trailing whitespace.

ful-stackz commented:
Looks good and ready to go! Thanks again!

ful-stackz commented:
@Zaid-Ajaj, actually can you please add some more stuff 😬 I can do it too, but I'm not sure how to push to your branch 😅 

- [ ] Add the same `.WithSummary()` API to
   - [ ] `ConstructorBuilder`
   - [ ] `FieldBuilder`
   - [ ] `InterfaceBuilder`
- [ ] Update the `CHANGELOG` with your contribution, something like `- Generating XML summary documentation for enums, enum members, interfaces, classes and class members ([Zaid Ajaj](https://github.com/Zaid-Ajaj))`
